README.md

```diff
@@ -1 +1,172 @@
+    - zero products is permitted
+- A `total` method which will calculate the total cost of the order by:
+  - summing up the products
+  - adding a 7.5% tax
+- An `add_product` method which will take in two parameters, product name and price, and add the data to the product collection
+  - It should return `true` if the item was succesfully added and `false` if it was not
+
+
+### Optional:
+Make sure to write tests for any optionals you implement!
+
+- Add a `remove_product` method to the `Order` class which will take in one parameter, a product name, and remove the product from the collection
+    - It should return `true` if the item was successfully remove and `false` if it was not
+- Create an new `Product` class in your `Grocery` module which will store the information about each product.
+  - Should include attributes like ID, price and quantity
+- Update the `Order` class to utilize objects from the `Product` class instead of the collection that is currently used
+
+## Wave 2
+
+### Learning Goals
+- Create and use class methods
+- Use a CSV file for loading data
+- Create your own tests to verify method correctness.
+
+### Requirements
+- Update the `Order` class to be able to handle all of the fields from the CSV file used as input
+  - To try it out, manually choose the data from the first line of the CSV file and ensure you can create a new instance of your `Order` using that data
+- Add the following **class** methods to your existing `Order` class
+  - `self.all` - returns a collection of `Order` instances, representing all of the Orders described in the CSV. See below for the CSV file specifications
+  - `self.find(id)` - returns an instance of `Order` where the value of the id field in the CSV matches the passed parameter.
+
+#### Error Handling
+- What should your program do if `Order.find` is called with an ID that doesn't exist?
+
+#### CSV Data File
+
+The data, in order in the CSV, consists of:  
+
+| Field    | Type     | Description
+|----------|----------|------------
+| ID       | Integer  | A unique identifier for that Order
+| Products  | String  | The list of products in the following format: `name:price;nextname:nextprice`
+
+### Optional:
+First, implement the optional requirement from Wave 1
+
+Then, add the following **class** methods to your existing `Product` class
+  - `self.all` - returns a collection of `Product` instances, representing all of the Products described in the CSV. See below for the CSV file specifications
+  - `self.find(id)` - returns an instance of `Product` where the value of the id field in the CSV matches the passed parameter
+
+#### CSV Data File
+The data for the products CSV file consists of:
+
+| Field          | Type    | Description
+|----------------|---------|------------
+| ID             | Integer | A unique identifier for that Product
+| Description | String | The product description
+| Price      | Float  | The product's price
+| Quantity     | Integer  | The product's quantity
```

Comment:
Should `Quantity` be part of the join table? Or are there going to be multiple entries for the same product in different orders?

If it's the second, we should explicitly tell them that, otherwise they will probably get confused.